### ISSUE TITLE
Inconsistent display and output of default field values 

### ISSUE THREAD

**Issue by Pezzab** [pods-framework/pods#5736]
## Issue Overview
I've created some plain text fields in a Pod that I want to populate with a default entry (to use as headings for sections so editors don't have to keep adding them but can change them if necessary).

Having set up the Pod and all of this was working as expected I added two extra fields with identical settings. Neither of the new default field entries appear in the post editor (though the original fields still work fine).

Yet the default field entries remain persisitent in the Pods editor.

## Expected Behavior
Default field entries should populate the field in the post editor unless overridden

## Current Behavior
Initial set up of fields with default entries worked fine. Additional fields with identical settings do not.

## Steps to Reproduce (for bugs)

1. Set up pod with plain text fields and default entry data
2. Work on site for a number of days without issue
3. Add new plain text fields to the same pod also with default entry data and identical settings
4. New default entry data does not display in post editor or can be output via template
5. Have duplicated issue between live and local site (behind firewall so cannot share link)

## Possible Solution
Suggested in Slack might be an issue with how pods data is being saved or overwritten

## WordPress Environment

Debug Information

WordPress Version: 5.4.1

PHP Version: 7.2.30-1+ubuntu18.04.1+deb.sury.org+1

MySQL Version: 5.5.5

Server Software: nginx/1.17.10

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:76.0) Gecko/20100101 Firefox/76.0

Session Save Path: /var/lib/php/sessions

Session Save Path Exists: Yes

Session Save Path Writeable: Yes

Session Max Lifetime: 1440

Opcode Cache:

    Apc: No
    Memcached: No
    OPcache: Yes
    Redis: No

Object Cache:

    APC: No
    APCu: No
    Memcache: Yes
    Memcached: Yes
    Redis: No

WPDB Prefix: wp_

WP Multisite Mode: No

WP Memory Limit: 40M

Current Memory Usage: 8.433M

Current Memory Usage (real): 2.000M

Pods Network-Wide Activated: No

Pods Install Location: /srv/www/qualsafe-academy-test/public_html/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: qualsafe-academy

Currently Active Plugins:

    Admin Columns: 4.1.6
    Contact Form 7: 5.1.9
    Pods - Custom Content Types and Fields: 2.7.20
    Query Monitor: 3.6.0
    WP Help: 1.7.0
    WP Migrate DB Pro: 1.9.10

## Pods Package Export (helpful!)
{"meta":{"version":"2.7.20","build":1591273413},"pods":{"148":{"id":148,"name":"course","label":"Courses","description":"","type":"post_type","storage":"meta","object":"","alias":"","fields":{"course_title_background_image":{"id":149,"name":"course_title_background_image","label":"Title background image","description":"","help":"","class":"","type":"file","weight":0,"pick_object":"custom-simple","pick_val":"","sister_id":"","required":"0","file_format_type":"single","file_uploader":"attachment","file_attachment_tab":"upload","file_edit_title":"1","file_show_edit_link":"0","file_linked":"0","file_limit":"0","file_restrict_filesize":"10MB","file_type":"images","file_field_template":"rows","file_add_button":"Add File","file_modal_title":"Attach a file","file_modal_add_button":"Add File","file_wp_gallery_output":"0","file_wp_gallery_link":"post","file_wp_gallery_columns":"1","file_wp_gallery_random_sort":"0","file_wp_gallery_size":"thumbnail","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","file_allowed_extensions":""},"course_description":{"id":165,"name":"course_description","label":"Course description","description":"Short description that appears at the top of the page","help":"","class":"","type":"paragraph","weight":1,"pick_object":"","pick_val":"","sister_id":"","required":"0","paragraph_allow_html":"1","paragraph_oembed":"0","paragraph_wptexturize":"1","paragraph_convert_chars":"1","paragraph_wpautop":"1","paragraph_allow_shortcode":"0","paragraph_allowed_html_tags":"strong em a ul ol li b i","paragraph_max_length":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","paragraph_repeatable":"0","unique":"0","paragraph_placeholder":""},"overview_section_heading":{"id":202,"name":"overview_section_heading","label":"Course Overview section heading","description":"Only Admin can change this heading","help":"","class":"","type":"text","weight":2,"pick_object":"","pick_val":"","sister_id":"","required":"0","text_allow_shortcode":"0","text_allow_html":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","pick_post_status":["publish"],"default_value":"Course Overview","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","text_repeatable":"0","text_placeholder":""},"course_overview":{"id":150,"name":"course_overview","label":"Course Overview","description":"","help":"","class":"","type":"wysiwyg","weight":3,"pick_object":"","pick_val":"","sister_id":"","required":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","wysiwyg_repeatable":"0","wysiwyg_allowed_html_tags":""},"course_image":{"id":151,"name":"course_image","label":"Course image","description":"","help":"","class":"","type":"file","weight":4,"pick_object":"custom-simple","pick_val":"","sister_id":"","required":"0","file_format_type":"single","file_uploader":"attachment","file_attachment_tab":"upload","file_edit_title":"1","file_show_edit_link":"0","file_linked":"0","file_limit":"0","file_restrict_filesize":"10MB","file_type":"images","file_field_template":"rows","file_add_button":"Add File","file_modal_title":"Attach a file","file_modal_add_button":"Add File","file_wp_gallery_output":"0","file_wp_gallery_link":"post","file_wp_gallery_columns":"1","file_wp_gallery_random_sort":"0","file_wp_gallery_size":"thumbnail","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","file_allowed_extensions":""},"whats_covered":{"id":152,"name":"whats_covered","label":"What's Covered?","description":"","help":"","class":"","type":"wysiwyg","weight":5,"pick_object":"","pick_val":"","sister_id":"","required":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","wysiwyg_repeatable":"0","wysiwyg_allowed_html_tags":""},"things_to_consider_heading":{"id":453,"name":"things_to_consider_heading","label":"Things to consider heading","description":"Only Admin can change this heading","help":"","class":"","type":"text","weight":6,"pick_object":"","pick_val":"","sister_id":"","required":"0","text_allow_shortcode":"0","text_allow_html":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","pick_post_status":["publish"],"default_value":"Things to consider before applying ","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","text_repeatable":"0","text_placeholder":""},"things_to_consider":{"id":384,"name":"things_to_consider","label":"Things to consider before applying","description":"Include required criteria and so on","help":"","class":"","type":"wysiwyg","weight":7,"pick_object":"","pick_val":"","sister_id":"","required":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","wysiwyg_repeatable":"0","wysiwyg_allowed_html_tags":""},"course_structure_heading":{"id":454,"name":"course_structure_heading","label":"Course structure heading","description":"","help":"","class":"","type":"text","weight":8,"pick_object":"","pick_val":"","sister_id":"","required":"0","text_allow_shortcode":"0","text_allow_html":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","pick_post_status":["publish"],"default_value":"Course structure","admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","text_repeatable":"0","text_placeholder":""},"course_structure":{"id":385,"name":"course_structure","label":"Course structure","description":"","help":"","class":"","type":"wysiwyg","weight":9,"pick_object":"","pick_val":"","sister_id":"","required":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","wysiwyg_repeatable":"0","wysiwyg_allowed_html_tags":""}},"show_in_menu":"1","label_singular":"Course","public":"1","show_ui":"1","supports_title":"1","supports_editor":"0","publicly_queryable":"1","exclude_from_search":"0","capability_type":"post","capability_type_custom":"course","capability_type_extra":"1","has_archive":"1","hierarchical":"0","rewrite":"1","rewrite_with_front":"1","rewrite_feeds":"0","rewrite_pages":"1","query_var":"1","can_export":"1","default_status":"draft","supports_author":"0","supports_thumbnail":"0","supports_excerpt":"0","supports_trackbacks":"0","supports_custom_fields":"0","supports_comments":"0","supports_revisions":"0","supports_page_attributes":"0","supports_post_formats":"0","built_in_taxonomies_category":"0","built_in_taxonomies_link_category":"0","built_in_taxonomies_post_tag":"0","menu_position":"0","show_in_nav_menus":"1","show_in_admin_bar":"1","pfat_enable":"0","pfat_run_outside_loop":"0","pfat_append_single":"replace","pfat_filter_single":"the_content","pfat_append_archive":"append","pfat_filter_archive":"the_content","rest_enable":"0","read_all":"0","write_all":"0","pfat_single":"Course single"}}}

These are the specific field headings I'm working on, the first two work as expected the second two don't:

Course Overview section heading
What's Covered section heading
Things to consider heading
Course structure heading

**Bear in mind that I set the pod up without the first two fields initially (so it might work fine with a clean install), the second two were added later when the bug occured**

**Comment by jimtrue** [comment]
![Screen Shot 2020-06-04 at 10 33 54 AM](https://user-images.githubusercontent.com/3052875/83770336-f2d61e00-a64e-11ea-9e7d-301e50e6ad14.png)
![Screen Shot 2020-06-04 at 10 33 34 AM](https://user-images.githubusercontent.com/3052875/83770337-f36eb480-a64e-11ea-9341-d8b73dc14b00.png)

Both of those default values are loading for me just fine. You've probably got a theme/plugin conflict if they're not showing up for you.

**Comment by Pezzab** [comment]
Out of interest where are the field defaults stored in the database - I was having a dig around in phpMyAdmin but couldn't figure out where to find them.

**Comment by Pezzab** [comment]
> 
> 
> Granted in my situation, this is a brand new configuration with no existing records. @Pezzab Are you indicating that the default values do not show on the edit pages of your existing records that are already saved?

That's correct. I already have identically formatted fields set up in the same Pod.

The ones I set up initially and the default values I assigned to them display and behave as expected.

Any additional fields I have set up in *exactly the same manner* just do not show the default values in the editor, or display them via my template, even though when I go back to edit teh Pod the value I've assigned is still there.

I'd like to sort this out manually if possible - because short of nuking the pods and starting again I'm not sure how to get around this.

By the way my dev work flow is a local dev site to a live test site:

All the content is set up on the live site then exported back to my local dev set up via Migrate WP Pro (using database download and import/export as the live push/pull feature won't work on my company's set up).

All code and Pods are developed locally and pushed to the live set up (so I do a copy paste migrate of Pods in this instance).

Just in case any of this has any bearing on the matter.
